### PR TITLE
chore: [IOBP-404] Update ID Pay API definitions for new exceptions

### PR DESCRIPTION
**Pull request by mastro993**
⚠️ Depends on https://github.com/pagopa/cstar-infrastructure/pull/1176 https://github.com/pagopa/io-dev-api-server/pull/327

## Short description
This PR updates the ID Pay definitions which includes new DTOs for error handling and the failure mappings.

## List of changes proposed in this pull request
- [package.json](https://github.com/pagopa/io-app/compare/IOBP-404-idpay-api-definitions-update?expand=1#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519): updated `idpay_api` definitions url;
- [ts/features/idpay/onboarding/xstate/services.ts](https://github.com/pagopa/io-app/compare/IOBP-404-idpay-api-definitions-update?expand=1#diff-ca9d15f03f5e50bc254f70f40bea5ce8dd76e54f1f073163d1fc2f6aedde6075): updated `onboardingErrorToFailure` mapping;
- [ts/features/idpay/payment/xstate/__tests__/services.test.ts](https://github.com/pagopa/io-app/compare/IOBP-404-idpay-api-definitions-update?expand=1#diff-19033164bbf27043f7ac25387954649f0091c3f980a21435b11dd025e719f5e2): updated tests

## How to test
Tests should solve correclty

### CODE REVIEW

File: ts/features/idpay/onboarding/screens/FailureScreen.tsx [hunk 1]
```diff
@@ -1,121 +1,171 @@
...
-          label={I18n.t("idpay.onboarding.failure.button.goToInitiative")}
-          accessibilityLabel={I18n.t(
-            "idpay.onboarding.failure.button.goToInitiative"
-          )}
-          onPress={handleNavigateToInitiativePress}
-          fullWidth={true}
-        />
-      );
+  // TODO add missing pictograms IOBP-176
+  const mapFailureToContentProps = (
```
Comment (forrest57) [inline ts/features/idpay/onboarding/screens/FailureScreen.tsx:49]: at this point, why not just handle this like this?
```ts
title: I18n.t(
            `idpay.onboarding.failure.message.${failure}.title`
          ),
subtitle: I18n.t(
            `idpay.onboarding.failure.message.${failure}.subtitle`
          ),
action: defaultCloseAction,
pictogram:myErrorPictograms[failure] // or just add it to I18n
```

Comment (mastro993) [reply ts/features/idpay/onboarding/screens/FailureScreen.tsx:49]: Not all configurations have title and/or subtitle and not alla configurations have the default close action. I tried a solution like the one you suggested but I found it a bit cumbersome.
Also, using a simple switch like this is very easy to mantain and edit if necessary, imho.
Finally, managing pictograms via i18n is not a good idea.

**Comment by forrest57** [comment]
`TSC` still fails because of the `ONBOARDED` status in a couple files; can you fix?

File: ts/components/screens/OperationResultScreenContent.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ import { Platform, SafeAreaView, StyleSheet, View } from "react-native";
 import { ScrollView } from "react-native-gesture-handler";
 import { LabelSmall } from "../core/typography/LabelSmall";
 
-type OperationResultScreenContent = {
+type OperationResultScreenContentProps = WithTestID<{
```
Comment (forrest57) [inline ts/components/screens/OperationResultScreenContent.tsx:18]: the type needs to be updated in the tests aswell

Comment (mastro993) [reply ts/components/screens/OperationResultScreenContent.tsx:18]: Fixed https://github.com/pagopa/io-app/pull/5224/commits/133129ab11764412c3c7bdd16215acb6b15a8648

**Review by forrest57 — APPROVED** [review]

**Pull request merged**